src/reducers/fileTree.tsx

```diff
@@ -372,6 +376,75 @@ export const goToRelativeFile = ({
   };
 };
 
+type GoToRelativeMessageParams = {
+  _getRelativeMessage?: typeof getRelativeMessage;
+  _viewVersionFile?: typeof viewVersionFile;
+  currentMessageUid: LinterMessage['uid'] | void;
+  currentPath: string;
+  messageMap: LinterMessageMap;
+  pathList: string[];
+  position: RelativePathPosition;
+  versionId: number;
+};
+
+export const goToRelativeMessage = ({
+  /* istanbul ignore next */
+  _getRelativeMessage = getRelativeMessage,
+  /* istanbul ignore next */
+  _viewVersionFile = viewVersionFile,
+  currentMessageUid,
+  currentPath,
+  messageMap,
+  pathList,
+  position,
+  versionId,
+}: GoToRelativeMessageParams): ThunkActionCreator => {
+  return async (dispatch, getState) => {
+    const nextRelativeMessageInfo = _getRelativeMessage({
+      currentMessageUid,
+      currentPath,
+      messageMap,
+      pathList,
+      position,
+    });
+
+    if (nextRelativeMessageInfo) {
+      const { line, path, uid } = nextRelativeMessageInfo;
+      const {
+        router: { location },
+      } = getState();
+
+      const queryParams = queryString.parse(location.search);
+      const newParams = { ...queryParams };
+      newParams.path = path;
+      newParams.messageUid = uid;
+
+      // Update the location with the hash for the message uid.
+      const newLocation = {
+        ...location,
+        hash: getCodeLineAnchor(line || 0),
```

Comment:
When `line` is empty, it means you're dealing with a global message so shouldn't this be `getCodeLineAnchor(line || 1)`? Or does `0` scroll to the top?

Reply:
Yes, in my next patch I am adding a `div` with an `id` that corresponds to `getCodeLineAnchor(0)` above the global linter messages to provide a destination for scrolling to global messages.

Reply:
That sounds good. Adding a test (if there's not one already) to make sure it goes to line 0 would be good.